File: frontend/src/types/maintenanceExecution.ts
/**
 * @file types/maintenanceExecution.ts
 * @description 维保执行跟踪类型定义 — 与后端实体字段严格对齐
 */

/** 维保执行主记录 */
export interface MaintenanceExecution {
  id: number;
  tenantId?: string;
  maintenanceRecordId: number;
  workOrderId: number;
  assigneeId: number | null;
  assigneeName: string | null;
  /** IDLE | RUNNING | PAUSED | COMPLETED */
  status: 'IDLE' | 'RUNNING' | 'PAUSED' | 'COMPLETED';
  totalLaborHours: number | null;
  totalMaterialCost: number | null;
  startTime: string | null;
  pauseTime: string | null;
  resumeTime: string | null;
  endTime: string | null;
  remark: string | null;
  createBy?: number;
  createTime?: string;
  updateTime?: string;
}

/** 施工步骤 */
export interface MaintenanceExecutionStep {
  id: number;
  tenantId?: string;
  executionId: number;
  stepName: string;
  stepOrder: number;
  description: string | null;
  operatorId: number | null;
  operatorName: string | null;
  laborHours: number | null;
  startTime: string | null;
  endTime: string | null;
  createTime?: string;
  updateTime?: string;
}

/** 物料/备件使用记录 */
export interface MaintenanceExecutionMaterial {
  id: number;
  tenantId?: string;
  executionId: number;
  materialName: string;
  specification: string | null;
  quantity: number;
  unitPrice: number | null;
  totalPrice: number | null;
  sourceWarehouse: string | null;
  remark: string | null;
  createTime?: string;
}

/** 开始施工请求 */
export interface ExecutionStartPayload {
  maintenanceRecordId: number;
  workOrderId: number;
  assigneeId?: number | null;
  assigneeName?: string | null;
  remark?: string | null;
}

/** 创建步骤请求 */
export interface StepCreatePayload {
  stepName: string;
  stepOrder?: number;
  description?: string | null;
  operatorId?: number | null;
  operatorName?: string | null;
  laborHours?: number | null;
}

/** 更新步骤请求 */
export interface StepUpdatePayload {
  stepName?: string;
  stepOrder?: number;
  description?: string | null;
  operatorId?: number | null;
  operatorName?: string | null;
  laborHours?: number | null;
}

/** 添加物料请求 */
export interface MaterialCreatePayload {
  materialName: string;
  specification?: string | null;
  quantity: number;
  unitPrice?: number | null;
  /** 合计金额（可选，后端会校验一致性） */
  totalPrice?: number | null;
  sourceWarehouse?: string | null;
  remark?: string | null;
}

File: backend/src/main/java/com/ams/entity/MaintenanceExecution.java
package com.ams.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 维保执行主记录实体。
 *
 * <p>维护施工执行生命周期（IDLE→RUNNING→PAUSED→RUNNING→COMPLETED），
 * 关联维保记录（maintenance_record）和工单（work_order）。
 *
 * <p>职责边界：本实体用于维保粒度的施工过程跟踪，
 * 与工单粒度步骤实体 WorkOrderStep 职责不同——前者记录维保施工步骤，
 * 后者记录工单内的检查清单步骤，两者共存但不冲突。
 */
@Data
@TableName("maintenance_execution")
public class MaintenanceExecution implements Serializable {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String tenantId;
    private Long maintenanceRecordId;
    private Long workOrderId;
    private Long assigneeId;
    private String assigneeName;

    /** 状态: IDLE/RUNNING/PAUSED/COMPLETED */
    private String status;

    private BigDecimal totalLaborHours;
    private BigDecimal totalMaterialCost;
    private LocalDateTime startTime;
    private LocalDateTime pauseTime;
    private LocalDateTime resumeTime;
    private LocalDateTime endTime;
    private String remark;
    private Long createBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    private Integer deleted;
}
